src/main/sensors/gyroanalyse.c

```diff
@@ -39,43 +39,41 @@
-#define FFT_SAMPLING_RATE            1000  // allows analysis up to 500Hz which is more than motors create
-#define FFT_BPF_HZ                    200  // use a bandpass on gyro data to ignore extreme low and extreme high frequencies
-#define DYN_NOTCH_WIDTH               100  // just an orientation and start value
-#define DYN_NOTCH_CHANGERATE           60  // lower cut does not improve the performance much, higher cut makes it worse...
-#define DYN_NOTCH_MIN_CUTOFF          120  // don't cut too deep into low frequencies
-#define DYN_NOTCH_MAX_CUTOFF          200  // don't go above this cutoff (better filtering with "constant" delay at higher center frequencies)
+#define FFT_WINDOW_SIZE       32  // max for f3 targets
+#define FFT_BIN_COUNT         (FFT_WINDOW_SIZE / 2)
+#define FFT_MIN_FREQ          100  // not interested in filtering frequencies below 100Hz
+#define FFT_SAMPLING_RATE     1000  // allows analysis up to 500Hz which is more than motors create
+#define FFT_MAX_FREQUENCY     (FFT_SAMPLING_RATE / 2) // nyquist rate
+#define FFT_BPF_HZ            200  // use a bandpass on gyro data to ignore extreme low and extreme high frequencies
+#define FFT_RESOLUTION        ((float)FFT_SAMPLING_RATE / FFT_WINDOW_SIZE) // hz per bin
+#define DYN_NOTCH_WIDTH       100  // just an orientation and start value
+#define DYN_NOTCH_CHANGERATE  60  // lower cut does not improve the performance much, higher cut makes it worse...
+#define DYN_NOTCH_MIN_CUTOFF  120  // don't cut too deep into low frequencies
+#define DYN_NOTCH_MAX_CUTOFF  200  // don't go above this cutoff (better filtering with "constant" delay at higher center frequencies)
 
 #define BIQUAD_Q 1.0f / sqrtf(2.0f)         // quality factor - butterworth
 
-static uint16_t samplingFrequency;          // gyro rate
-static uint8_t fftBinCount;
-static float fftResolution;                 // hz per bin
-static float gyroData[3][FFT_WINDOW_SIZE];  // gyro data used for frequency analysis
+static FAST_RAM uint16_t fftSamplingScale;
 
-static arm_rfft_fast_instance_f32 fftInstance;
-static float fftData[FFT_WINDOW_SIZE];
-static float rfftData[FFT_WINDOW_SIZE];
-static gyroFftData_t fftResult[3];
-static uint16_t fftMaxFreq = 0;             // nyquist rate
-static uint16_t fftIdx = 0;                 // use a circular buffer for the last FFT_WINDOW_SIZE samples
+// gyro data used for frequency analysis
+static float FAST_RAM gyroData[XYZ_AXIS_COUNT][FFT_WINDOW_SIZE];
 
+static FAST_RAM arm_rfft_fast_instance_f32 fftInstance;
+static FAST_RAM float fftData[FFT_WINDOW_SIZE];
+static FAST_RAM float rfftData[FFT_WINDOW_SIZE];
+static FAST_RAM gyroFftData_t fftResult[3];
 
-// accumulator for oversampled data => no aliasing and less noise
-static float fftAcc[3] = {0, 0, 0};
-static int fftAccCount = 0;
-static int fftSamplingScale;
+// use a circular buffer for the last FFT_WINDOW_SIZE samples
+static FAST_RAM uint16_t fftIdx = 0;
```

Comment:
This `= 0` doesn't make sense here, as these variables are initialized to 0 in ResetHandler.
Moreover, only trivial initialization is supported for now, if you write `= 1` in here, ResetHandler will still initialize to 0 :-)

Reply:
Ok, this is just as was in untouched code. I can change it to no initialization.

Reply:
Just for uniformity :)